Error from the pod log without providing the system property:

```
2023-10-19T13:29:37.3291228Z WARN: example-count0-kc-99ndn not ready log: start(DefaultCacheManager.java:747)
2023-10-19T13:29:37.3292598Z 	at org.infinispan.manager.DefaultCacheManager.<init>(DefaultCacheManager.java:411)
2023-10-19T13:29:37.3294579Z 	at org.keycloak.quarkus.runtime.storage.legacy.infinispan.CacheManagerFactory.startCacheManager(CacheManagerFactory.java:96)
2023-10-19T13:29:37.3296408Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2023-10-19T13:29:37.3298126Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2023-10-19T13:29:37.3299710Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2023-10-19T13:29:37.3300884Z 	at java.base/java.lang.Thread.run(Thread.java:840)
2023-10-19T13:29:37.3311314Z Caused by: org.infinispan.commons.CacheConfigurationException: ISPN000541: Error while trying to create a channel using the specified configuration '[TCP(bundler.max_size=64000, sock_conn_timeout=300, port_range=0, thread_pool.keep_alive_time=60000, diag.enabled=false, bind_port=7800, thread_naming_pattern=pl, thread_pool.thread_dumps_threshold=10000, send_buf_size=640k, thread_pool.max_threads=200, bundler_type=transfer-queue, logical_addr_cache_expiration=360000, thread_pool.min_threads=0), RED(), dns.DNS_PING(num_discovery_runs=3), MERGE3(max_interval=30000, min_interval=10000), FD_SOCK2(offset=50000), FD_ALL3(), VERIFY_SUSPECT2(timeout=1000), pbcast.NAKACK2(xmit_table_num_rows=50, use_mcast_xmit=false, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000, xmit_interval=200, resend_last_seqno=true), UNICAST3(conn_close_timeout=5000, xmit_interval=200, xmit_table_num_rows=50, xmit_table_msgs_per_row=1024, xmit_table_max_compaction_time=30000), pbcast.STABLE(desired_avg_gossip=5000, max_bytes=1M), pbcast.GMS(join_timeout=2000, print_local_addr=false), UFC(min_threshold=0.40, max_credits=4m), MFC(min_threshold=0.40, max_credits=4m), FRAG4(frag_size=60000)]'
2023-10-19T13:29:37.3321415Z 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.channelFromConfigurator(JGroupsTransport.java:779)
2023-10-19T13:29:37.3323476Z 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.buildChannel(JGroupsTransport.java:748)
2023-10-19T13:29:37.3325390Z 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.initChannel(JGroupsTransport.java:503)
2023-10-19T13:29:37.3327207Z 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.start(JGroupsTransport.java:484)
2023-10-19T13:29:37.3328950Z 	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:42)
2023-10-19T13:29:37.3330609Z 	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:27)
2023-10-19T13:29:37.3332465Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:616)
2023-10-19T13:29:37.3334558Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:607)
2023-10-19T13:29:37.3336638Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:576)
2023-10-19T13:29:37.3338775Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:807)
2023-10-19T13:29:37.3340563Z 	at org.infinispan.metrics.impl.MetricsCollector.start(MetricsCollector.java:84)
2023-10-19T13:29:37.3341922Z 	at org.infinispan.metrics.impl.CorePackageImpl$1.start(CorePackageImpl.java:41)
2023-10-19T13:29:37.3343270Z 	at org.infinispan.metrics.impl.CorePackageImpl$1.start(CorePackageImpl.java:34)
2023-10-19T13:29:37.3345016Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:616)
2023-10-19T13:29:37.3457909Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:607)
2023-10-19T13:29:37.3460292Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:576)
2023-10-19T13:29:37.3462441Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:807)
2023-10-19T13:29:37.3464640Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startDependencies(BasicComponentRegistryImpl.java:634)
2023-10-19T13:29:37.3466985Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:598)
2023-10-19T13:29:37.3469522Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:576)
2023-10-19T13:29:37.3471658Z 	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:807)
2023-10-19T13:29:37.3473728Z 	at org.infinispan.factories.AbstractComponentRegistry.internalStart(AbstractComponentRegistry.java:379)
2023-10-19T13:29:37.3475695Z 	at org.infinispan.factories.AbstractComponentRegistry.start(AbstractComponentRegistry.java:252)
2023-10-19T13:29:37.3477377Z 	at org.infinispan.manager.DefaultCacheManager.internalStart(DefaultCacheManager.java:779)
2023-10-19T13:29:37.3478445Z 	... 7 more
2023-10-19T13:29:37.3479200Z Caused by: java.lang.IllegalArgumentException: dns_query can not be null or empty
2023-10-19T13:29:37.3480472Z 	at org.jgroups.protocols.dns.DNS_PING.validateProperties(DNS_PING.java:85)
```

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
@@ -65,6 +65,8 @@ import static org.keycloak.operator.crds.v2alpha1.CRDUtils.isTlsConfigured;
 @KubernetesDependent(labelSelector = Constants.DEFAULT_LABELS_AS_STRING)
 public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependentResource<StatefulSet, Keycloak> {
 
+    static final String JGROUPS_DNS_QUERY_PARAM = "-Djgroups.dns.query=";
+
     public static final String OPTIMIZED_ARG = "--optimized";
 
     @Inject
@@ -209,6 +211,7 @@ public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependent
         if (customImage.isPresent()) {
             containerBuilder.addToArgs(OPTIMIZED_ARG);
         }
+        containerBuilder.addToArgs(0, getJGroupsParameter(keycloakCR));
 
         // probes
         var tlsConfigured = isTlsConfigured(keycloakCR);
@@ -269,6 +272,10 @@ public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependent
             .endContainer().endSpec().endTemplate().endSpec().build();
 
         return baseDeployment;
+    }
+
+    private static String getJGroupsParameter(Keycloak keycloakCR) {
+        return JGROUPS_DNS_QUERY_PARAM + KeycloakDiscoveryServiceDependentResource.getName(keycloakCR) +"." + keycloakCR.getMetadata().getNamespace();
     }
 
     private void addEnvVarsAndWatchSecrets(StatefulSet baseDeployment, Keycloak keycloakCR) {
@@ -349,12 +356,6 @@ public class KeycloakDeploymentDependentResource extends CRUDKubernetesDependent
                         .withOptional(false)
                         .endSecretKeyRef()
                         .endValueFrom()
-                        .build());
-
-        envVars.add(
-                new EnvVarBuilder()
-                        .withName("jgroups.dns.query")
-                        .withValue(getName(keycloakCR) + Constants.KEYCLOAK_DISCOVERY_SERVICE_SUFFIX +"." + keycloakCR.getMetadata().getNamespace())
                         .build());
 
         return envVars;

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJobDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJobDependentResource.java
@@ -120,8 +120,11 @@ public class KeycloakRealmImportJobDependentResource extends KubernetesDependent
 
         var runBuild = !keycloakContainer.getArgs().contains(KeycloakDeploymentDependentResource.OPTIMIZED_ARG) ? "/opt/keycloak/bin/kc.sh --verbose build && " : "";
 
+        String jgroupsDNSQuery = keycloakContainer.getArgs().stream()
+                .filter(s -> s.startsWith(KeycloakDeploymentDependentResource.JGROUPS_DNS_QUERY_PARAM))
+                .reduce((s1, s2) -> s2).map(" "::concat).orElse("");
         var commandArgs = List.of("-c",
-                runBuild + "/opt/keycloak/bin/kc.sh --verbose import --optimized --file='" + importMntPath + realmName + "-realm.json' " + override);
+                runBuild + "/opt/keycloak/bin/kc.sh --verbose import --optimized --file='" + importMntPath + realmName + "-realm.json' " + override + jgroupsDNSQuery);
 
         keycloakContainer.setCommand(command);
         keycloakContainer.setArgs(commandArgs);

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJobDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJobDependentResource.java
@@ -120,11 +120,8 @@ public class KeycloakRealmImportJobDependentResource extends KubernetesDependent
 
         var runBuild = !keycloakContainer.getArgs().contains(KeycloakDeploymentDependentResource.OPTIMIZED_ARG) ? "/opt/keycloak/bin/kc.sh --verbose build && " : "";
 
-        String jgroupsDNSQuery = keycloakContainer.getArgs().stream()
-                .filter(s -> s.startsWith(KeycloakDeploymentDependentResource.JGROUPS_DNS_QUERY_PARAM))
-                .reduce((s1, s2) -> s2).map(" "::concat).orElse("");
         var commandArgs = List.of("-c",
-                runBuild + "/opt/keycloak/bin/kc.sh --verbose import --optimized --file='" + importMntPath + realmName + "-realm.json' " + override + jgroupsDNSQuery);
+                runBuild + "/opt/keycloak/bin/kc.sh --verbose import --optimized --file='" + importMntPath + realmName + "-realm.json' " + override);
 
         keycloakContainer.setCommand(command);
         keycloakContainer.setArgs(commandArgs);

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJobDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJobDependentResource.java
@@ -38,6 +38,7 @@ import org.keycloak.operator.Utils;
 import org.keycloak.operator.crds.v2alpha1.realmimport.KeycloakRealmImport;
 
 import java.util.List;
+import java.util.Set;
 
 import static org.keycloak.operator.controllers.KeycloakDistConfigurator.getKeycloakOptionEnvVarName;
 
@@ -76,7 +77,9 @@ public class KeycloakRealmImportJobDependentResource extends KubernetesDependent
 
         var cacheEnvVarName = getKeycloakOptionEnvVarName("cache");
         var healthEnvVarName = getKeycloakOptionEnvVarName("health-enabled");
-        envvars.removeIf(e -> e.getName().equals(cacheEnvVarName) || e.getName().equals(healthEnvVarName));
+        var cacheStackEnvVarName = getKeycloakOptionEnvVarName("cache-stack");
+        var toRemove = Set.of(cacheEnvVarName, healthEnvVarName, cacheStackEnvVarName);
+        envvars.removeIf(e -> toRemove.contains(e.getName()));
 
         // The Job should not connect to the cache
         envvars.add(new EnvVarBuilder().withName(cacheEnvVarName).withValue("local").build());

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
@@ -319,6 +319,8 @@ public class BaseOperatorTest implements QuarkusTestAfterEachCallback {
               logFailed(k8sclient.apps().deployments().withName("keycloak-operator"), Deployment::getStatus);
           }
           logFailed(k8sclient.apps().statefulSets().withName(POSTGRESQL_NAME), StatefulSet::getStatus);
+          k8sclient.pods().withLabel("app", "keycloak-realm-import").list().getItems().stream()
+                  .forEach(pod -> logFailed(k8sclient.pods().resource(pod), Pod::getStatus));
       } finally {
           cleanup();
       }

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java
@@ -489,7 +489,7 @@ public class KeycloakDeploymentTest extends BaseOperatorTest {
                 .list()
                 .getItems();
 
-        assertThat(pods.get(0).getSpec().getContainers().get(0).getArgs()).containsExactly("--verbose", "start", "--optimized");
+        assertThat(pods.get(0).getSpec().getContainers().get(0).getArgs()).endsWith("--verbose", "start", "--optimized");
     }
 
     @Test
@@ -512,7 +512,7 @@ public class KeycloakDeploymentTest extends BaseOperatorTest {
                 .list()
                 .getItems();
 
-        assertThat(pods.get(0).getSpec().getContainers().get(0).getArgs()).containsExactly("--verbose", "start", "--optimized");
+        assertThat(pods.get(0).getSpec().getContainers().get(0).getArgs()).endsWith("--verbose", "start", "--optimized");
         assertThat(pods.get(0).getSpec().getImagePullSecrets().size()).isEqualTo(1);
         assertThat(pods.get(0).getSpec().getImagePullSecrets().get(0).getName()).isEqualTo(imagePullSecretName);
     }

--- a/operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
@@ -194,8 +194,8 @@ public class PodTemplateTest {
         // Assert
         assertEquals(1, podTemplate.getSpec().getContainers().get(0).getCommand().size());
         assertEquals(command, podTemplate.getSpec().getContainers().get(0).getCommand().get(0));
-        assertEquals(1, podTemplate.getSpec().getContainers().get(0).getArgs().size());
-        assertEquals(arg, podTemplate.getSpec().getContainers().get(0).getArgs().get(0));
+        assertEquals(2, podTemplate.getSpec().getContainers().get(0).getArgs().size());
+        assertEquals(arg, podTemplate.getSpec().getContainers().get(0).getArgs().get(1));
     }
 
     @Test

